components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/utils/mappings/APIMappingUtil.java

```diff
@@ -550,6 +550,24 @@ public static APIListDTO fromAPIListToDTO(List<API> apiList, int offset, int lim
         return apiListDTO;
     }
 
+    /**
+     * Converts a List object of APIs into a DTO
+     *
+     * @param apiList List of APIs
+     * @return APIListDTO object containing APIDTOs
+     */
+    public static APIListDTO fromAPIListToDTO(List<API> apiList) {
+        APIListDTO apiListDTO = new APIListDTO();
+        List<APIInfoDTO> apiInfoDTOs = apiListDTO.getList();
+        if(apiList !=null) {
+            for (int i = 0; i <= apiList.size() - 1; i++) {
```

Comment:
Any reason for not using a "for each" style loop in this case? Its a lot less simpler and less error prone